### PR TITLE
fix(run_launcher): bump scaleway sdk with breaking changes

### PR DESCRIPTION
**Pull request by cyclimse**
## Changes
- Bumped scaleway-sdk-go to v2.1.0
- Fixed issue with breaking changes regarding the `/start` endpoint for JobDefinitions. It now can return multiple job runs. The returned `run-id` was used to identify the Dagster run from the job run. It's necessary for the health check to work.
- Removed the resume functionality

## What do we need it?
- Currently, this project is not functional due to the use of the old SDK without the breaking changes on the /start endpoint
- As a result, the [tutorial](https://www.scaleway.com/en/docs/tutorials/dagster-serverless-jobs/) is not working as intended

## How have you tested it?

Tested locally by following the steps of the tutorial (https://www.scaleway.com/en/docs/tutorials/dagster-serverless-jobs/)

image

My only change was using `dagster-scaleway` from my MR.

I also tested changing the `cpu_limit` in the Dagster config, the job definition gets updated with the new value.

## Note
For reviewers, the CI injects the version directly, so I don't need to change it.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ dagster-scaleway = "dagster_scaleway.cli:main"
 
 [tool.poetry.dependencies]
 python = "^3.11"
-scaleway = "^1.4.1"
+scaleway = "^2.0.0"
```
Comment (cyclimse) [inline pyproject.toml]: note: do not merge as this, there's an issue with `list_job_definitions_all` with this version of the sdk

File: dagster_scaleway/serverless_job_launcher.py [hunk 2]
```diff
@@ -223,25 +244,11 @@ def launch_run(self, context: LaunchRunContext) -> None:
             instance_ref=self._instance.get_ref(),
         ).get_command_args()
 
-        self._launch_serverless_job_with_command(run, docker_image, command)
+        self._start_serverless_job_with_command(run, docker_image, list(command))
 
     @property
     def supports_resume_run(self):
```
Comment (cyclimse) [inline dagster_scaleway/serverless_job_launcher.py:252]: note: for reviewers, I've removed this feature as I haven't properly tested it.

File: dagster_scaleway/serverless_job_launcher.py [hunk 3]
```diff
@@ -303,18 +310,18 @@ def check_run_worker_health(self, run: DagsterRun):
...
+            status=SERVERLESS_JOBS_STATES_TO_WORKER_STATUS.get(
+                job_run.state, WorkerStatus.UNKNOWN
+            ),
+            run_worker_id=run.run_id,
+            msg=job_run.error_message,
+            transient=job_run.state in scw.JOB_RUN_TRANSIENT_STATUSES,
         )
 
-        if job_run.error_message:
-            health.msg = job_run.error_message
```
Comment (cyclimse) [inline dagster_scaleway/serverless_job_launcher.py:318]: I don't know I just missed it or there were some changes done to typing in dagster and/or the SDK, but both `msg` and `error_message` are nullable strings `None | str`

File: pyproject.toml (hunk 1 above)
Comment (cyclimse) [reply pyproject.toml]: fixed in release v2.1.0

File: dagster_scaleway/serverless_job_context.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ class ScalewayServerlessJobContext(
         [
             ("docker_image", Optional[str]),
             ("env_vars", Sequence[str]),
-            ("region", str),
+            ("region", Optional[str]),
```
Comment (cyclimse) [inline dagster_scaleway/serverless_job_context.py:60]: note: Region was already optional, this is a miss in the type hint

File: dagster_scaleway/serverless_job_launcher.py [hunk 5]
```diff
@@ -223,25 +246,11 @@ def launch_run(self, context: LaunchRunContext) -> None:
             instance_ref=self._instance.get_ref(),
         ).get_command_args()
 
-        self._launch_serverless_job_with_command(run, docker_image, command)
+        self._start_serverless_job_with_command(run, docker_image, list(command))
```
Comment (redanrd) [inline dagster_scaleway/serverless_job_launcher.py:249]: question: out of curiosity, is the renaming from `_launch` to `_start` only to comply with Serverless Jobs jargon (start), or for another reason?

**Review by redanrd — APPROVED** [review]

File: dagster_scaleway/serverless_job_launcher.py (hunk 5 above)
Comment (cyclimse) [reply dagster_scaleway/serverless_job_launcher.py:249]: yes, good point! When reading the code once again, I did not like I was using a different language than the API, so I changed it to be more consistent. 

I think you "launch" a Dagster run but you "start" a Scaleway Serverless Job definition and I got the two confused :smile:

**Review by Shillaker — APPROVED** [review]

**Pull request merged**